spyder/widgets/mixins.py

```diff
@@ -971,14 +1017,24 @@ def is_space(move):
                 return not to_text_string(curs.selectedText()).strip()
 
             def is_special_character(move):
-                    curs = self.textCursor()
-                    curs.movePosition(move, QTextCursor.KeepAnchor)
-                    text_cursor = to_text_string(curs.selectedText()).strip()
-                    return len(re.findall(r'([^\d\W]\w*)',
-                                          text_cursor, re.UNICODE)) == 0
+                curs = self.textCursor()
+                curs.movePosition(move, QTextCursor.KeepAnchor)
+                text_cursor = to_text_string(curs.selectedText()).strip()
+                return len(
+                    re.findall(r'([^\d\W]\w*)', text_cursor, re.UNICODE)) == 0
+
+            def is_dot(move):
+                curs = self.textCursor()
+                curs.movePosition(move, QTextCursor.KeepAnchor)
+                text_cursor = to_text_string(curs.selectedText()).strip()
+                return text_cursor == '.'
+
             if help_req:
-                if is_special_character(QTextCursor.NoMove):
+                if (is_special_character(QTextCursor.NoMove) and
+                        not is_dot(QTextCursor.PreviousCharacter)):
                     cursor.movePosition(QTextCursor.WordLeft)
+                elif is_dot(QTextCursor.PreviousCharacter):
+                    cursor.movePosition(QTextCursor.NextCharacter)

```

Comment:
Checking for `dot` here makes things specialized for Python. What's the plan for other languages?